sententia/src/main/java/nl/nfi/sententia/SententiaPlugin.java

```diff
@@ -170,10 +171,13 @@ public void domainObjectChanged(DomainObjectChangedEvent event) {
 		Iterator<DomainObjectChangeRecord> it = event.iterator();
 		while (it.hasNext()) {
 			DomainObjectChangeRecord evt = it.next();
-			if (evt.getEventType().getId() == ProgramEvent.SYMBOL_RENAMED.getId()) {				
+			// Check if we're dealing with a rename and the is applied to a function
+			if (evt.getEventType().getId() == ProgramEvent.SYMBOL_RENAMED.getId() && ((ProgramChangeRecord)evt).getObject() instanceof FunctionSymbol) {				
+				
+				Function changedFunction = currentProgram.getFunctionManager().getFunctionAt(((ProgramChangeRecord)evt).getStart());
 				
 				try {
-					FunctionDescriptor changedFunctionDescriptor = new FunctionDescriptor(currentProgram.getFunctionManager().getFunctionAt(((ProgramChangeRecord)evt).getStart()));
+					FunctionDescriptor changedFunctionDescriptor = new FunctionDescriptor(changedFunction);
 					sententiaAPI.addSignatureToDB(changedFunctionDescriptor);
```

Comment:
We could ask ourselves whether the additional local makes things more readable, would this make sense in a single go?


```suggestion
					sententiaAPI.addSignatureToDB(new FunctionDescriptor(changedFunction));
```